### ISSUE TITLE
Possible incorrect use of exported resource

### ISSUE THREAD

**Issue by nick-fox** [corey-hammerton/puppet-metricbeat#44]
Exported resource is causing duplicate resource. 

`Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: A duplicate resource was found while collecting exported resources, with the type and title Metricbeat::Modulesd[system]` 

 _An exported resource declaration specifies a desired state for a resource, and publishes the resource for use by other nodes. It does not manage the resource on the target system. Any node, including the node that exports it, can collect the exported resource and manage its own copy of it._
[https://puppet.com/docs/puppet/latest/lang_exported.html](See)

https://github.com/corey-hammerton/puppet-metricbeat/blob/0b9c772e1f8f0471511999d8b6a26e448bb343f6/manifests/config.pp#L73-L78

I think you should be using create_resources
_Converts a hash into a set of resources and adds them to the catalog._
[https://puppet.com/docs/puppet/latest/function.html#createresources](See)

**Comment by efoft** [comment]
Agree, I hit the same issue. If more than one node applies this manifest within the same puppetserver.

**Comment by braddeicide** [comment]
+1

**Comment by rogermartensson** [comment]
Is this fix planned to be included in a future version?
I noticed that the PR is failing.

**Comment by smoeding** [comment]
It's probably sufficient to turn the double `@@` into a single `@` here.

As already mentioned the double `@@` creates exported resources that can be picked up by other nodes. I don't see any use for node2 to create a metricbeat config entry that node1 created.

Using a single `@` will create virtual resources. These virtual resources must be collected (done here using `<||>`) to be realized. Now this could actually make sense here. A profile or role class could create virtual resources for items managed by the profile or role. These virtual resources would be ignored on servers that do not also include metricbeat. But if metricbeat is included then these resources would also be realized and added to the metricbeat config.

Obviously removing the `@@` would directly create the resources and the collection would not be needed.

**Comment by couloum** [comment]
Hi,

I had the same issue while trying to use this module on multiple nodes. I applied @smoeding fix and it worked like a charm.

Is the fix planned for a future release?

Thanks

**Comment by cdenneen** [comment]
@corey-hammerton this is a pretty big issue. Any solution for fixing this?

**Comment by cdenneen** [comment]
#59 added with @smoeding suggestion

**Comment by cdenneen** [comment]
@corey-hammerton Either suggestion is fine #48 or @smoeding that I did in #59 (it doesn't collect and realize those virtual resources so this would be done outside this class)

**Comment by corey-hammerton** [comment]
I merged #59 but am unable to verify the fix. Travis CI integration is now broken, this was unverified because this project is not under active development.
